### PR TITLE
Record embargo lift

### PR DESCRIPTION
**Pull request by philippgualdi**
Lift an embargo in a record with a scheduled task

### CODE REVIEW

File: invenio_records_marc21/services/services.py [hunk 1]
```diff
@@ -85,6 +90,47 @@ def update_draft(
...
         return super().update_draft(id_, identity, data, revision_id)
 
+    def _lift_embargo_from(self, record):
+        """Lifts embargo from record or draft."""
+        if not record.access.embargo.lift():
+            raise EmbargoNotLiftedError(record["id"])
+        record.access.protection.metadata = "public"
+        record.access.protection.files = "public"
+
+    def _draft_access_field_was_modified(self, draft, record):
```
Comment (utnapischtim) [inline invenio_records_marc21/services/services.py:100]: the name does not suggest, that the return type is a bool. i would prefer "_is_draft_access_field_modified"

File: invenio_records_marc21/services/services.py [hunk 2]
```diff
@@ -85,6 +90,47 @@ def update_draft(
         data = self._create_data(identity, data, metadata, access)
         return super().update_draft(id_, identity, data, revision_id)
 
+    def _lift_embargo_from(self, record):
+        """Lifts embargo from record or draft."""
+        if not record.access.embargo.lift():
```
Comment (utnapischtim) [inline invenio_records_marc21/services/services.py:95]: where is the lift method defined?

File: invenio_records_marc21/services/services.py (hunk 1 above)
Comment (philippgualdi) [reply invenio_records_marc21/services/services.py:100]: Ok

File: invenio_records_marc21/services/services.py (hunk 2 above)
Comment (philippgualdi) [reply invenio_records_marc21/services/services.py:95]: The lift function is here  [https://github.com/tu-graz-library/invenio-records-marc21/blob/master/invenio_records_marc21/records/systemfields/access/embargo.py#L45](here) defined.

File: invenio_records_marc21/services/services.py [hunk 3]
```diff
@@ -85,6 +90,47 @@ def update_draft(
...
+    def _draft_access_field_was_modified(self, draft, record):
+        """Returns True if draft's access field was modified."""
+        return draft.get('access') == record.get('access')
+
+    def lift_embargo(self, _id, identity):
+        """Lifts embargo from the record and updates draft."""
+        # Get the record
+        record = self.record_cls.pid.resolve(_id)
+
+        # Check permissions
```
Comment (utnapischtim) [inline invenio_records_marc21/services/services.py:109]: the comment is somehow misleadingly, a check indicates that the function returns true or false, but than the function call would be useless. the definition of the function indicates that if the sub function call returns false it throws a error. Better to remove the comment

File: invenio_records_marc21/services/services.py [hunk 4]
```diff
@@ -85,6 +90,47 @@ def update_draft(
...
+        self.require_permission(identity, "lift_embargo", record=record)
+
+        lifted_embargo_from_draft = False
+        # Check if record has already a draft
+        if record.has_draft:
+            draft = self.draft_cls.pid.resolve(_id, registered_only=False)
+            # If the draft has no modifications in the access field the
+            # embargo is lifted
+            if self._draft_access_field_was_modified(draft, record):
+                # Lifts embargo from draft
```
Comment (utnapischtim) [inline invenio_records_marc21/services/services.py:119]: remove that comment, it does not say more then the function call itself

File: invenio_records_marc21/services/services.py [hunk 5]
```diff
@@ -85,6 +90,47 @@ def update_draft(
...
+                self._lift_embargo_from(draft)
+                lifted_embargo_from_draft = True
+
+        # Lifts embargo from record
+        self._lift_embargo_from(record)
+        # Commit and index
+        record.commit()
```
Comment (utnapischtim) [inline invenio_records_marc21/services/services.py:126]: is there no function that does what is written in that comment? to commit all changes and reindex?

**Review by utnapischtim — APPROVED** [review]

**Pull request merged**